rat/variables.py

```diff
@@ -95,6 +106,27 @@ def get_numpy_indices(self, df):
             "__index"
         ].to_numpy(dtype=int)
 
+    def log_summary(self, log_level=logging.INFO):
+        for index_num in range(len(self.indexed_sets)):
+            logging.log(log_level, f"Subscript {index_num} - defined as union of the following columns({','.join(self.indexed_sets[index_num])}), with column name '{self.df.columns[index_num]}'")
+
+    def check_and_return_index(self, index_key: Tuple[str]) -> Tuple[str]:
```

Comment:
@bbbales2 
This commit added this function, where I intended to make `variables.Index` keep track of the subscripts and dataframe column names. For example for the baseball model, we subscript `skills` in 3 different ways (`skills[home_team, year]`, `skills[away_team, year]`, `skills[team, year]`), which kinda confuses what names we should set the underlying df(`self.df`)to be and when loc-ing them. 
Previously on the original implementation, we just used the column names of the first indexing call, but actually keeping track the different combinations of how the subscripts called gives us better opportunity for maintenance and improvement.

On an added plus, it allows for some cool info output:
```
INFO:root:rat compiler - starting code generation for 6 line(s) of code.
INFO:root:Subscript data for parameter skills:
INFO:root:Subscript 0 - defined as union of the following columns(home_team,away_team), with column name 'team'
INFO:root:Subscript 1 - defined as union of the following columns(year), with column name 'year'
INFO:root:-----
INFO:root:Subscript data for parameter sigma_mod_plus:
INFO:root:Subscript 0 - defined as union of the following columns(year), with column name 'year'
INFO:root:-----
INFO:root:Subscript data for parameter skills_mu:
INFO:root:Subscript 0 - defined as union of the following columns(year), with column name 'year'
INFO:root:-----
INFO:root:Subscript data for parameter sigma_mod:
INFO:root:Subscript 0 - defined as union of the following columns(year), with column name 'year'
INFO:root:-----
INFO:root:Subscript data for parameter sigma:
INFO:root:Subscript 0 - defined as union of the following columns(year), with column name 'year'
INFO:root:-----
```

Reply:
Looks good!